driver-core/src/main/java/com/datastax/driver/core/DataTypeCqlNameParser.java

```diff
@@ -148,7 +148,7 @@ static DataType parse(String toParse, Cluster cluster, String currentKeyspaceNam
         if (userType == null)
             throw new UnresolvedUserTypeException(currentKeyspaceName, type);
         else
-            return userType;
+            return userType.copy(frozen);
```

Comment:
I think it happens if, as part of a full keyspace refresh, you've already parsed the definition of `udt1` (it's in `currentUserTypes` but not frozen), and now you're parsing a table that has a `frozen<udt1>` column.